pkg/storage/mongo/getter.go

```diff
@@ -56,15 +59,17 @@ func (s *ModuleStore) Zip(ctx context.Context, module, vsn string) (io.ReadClose
 	defer span.End()
 
 	zipName := s.gridFileName(module, vsn)
-	fs := s.s.DB(s.d).GridFS("fs")
-	f, err := fs.Open(zipName)
+	db := s.client.Database(s.db)
+	bucket, err := gridfs.NewBucket(db, &options.BucketOptions{})
 	if err != nil {
-		kind := errors.KindUnexpected
-		if err == mgo.ErrNotFound {
-			kind = errors.KindNotFound
-		}
-		return nil, errors.E(op, err, kind, errors.M(module), errors.V(vsn))
+		return nil, errors.E(op, err, errors.M(module), errors.V(vsn))
+	}
+
+	dStream, err := bucket.OpenDownloadStreamByName(zipName, options.GridFSName())
+	defer dStream.Close()
```

Comment:
We should not close the reader here otherwise we would try to read from a closed reader later on.